crates/aptos-rosetta-cli/src/construction.rs

```diff
@@ -273,3 +275,61 @@ impl SetVoterCommand {
             .await
     }
 }
+
+/// Initialize stake amount
+///
+///
+#[derive(Debug, Parser)]
+pub struct CreateStakePoolCommand {
+    #[clap(flatten)]
+    network_args: NetworkArgs,
+    #[clap(flatten)]
+    url_args: UrlArgs,
+    #[clap(flatten)]
+    encoding_options: EncodingOptions,
+    #[clap(flatten)]
+    profile_options: ProfileOptions,
+    #[clap(flatten)]
+    private_key_options: PrivateKeyInputOptions,
+    #[clap(flatten)]
+    txn_args: TransactionArgs,
+    /// The sending account, since the private key doesn't always match the
+    /// AccountAddress if it rotates
+    #[clap(long, parse(try_from_str=aptos::common::types::load_account_arg))]
+    sender: Option<AccountAddress>,
+    /// Operator
+    #[clap(long, parse(try_from_str=aptos::common::types::load_account_arg))]
+    operator: Option<AccountAddress>,
+    /// Voter
+    #[clap(long, parse(try_from_str=aptos::common::types::load_account_arg))]
+    voter: Option<AccountAddress>,
+    /// Amount
+    #[clap(long)]
+    amount: Option<u64>,
+}
+
+impl CreateStakePoolCommand {
+    pub async fn execute(self) -> anyhow::Result<TransactionIdentifier> {
+        info!("CreateStakePool {:?}", self);
+        let client = self.url_args.client();
+        let network_identifier = self.network_args.network_identifier();
+        let private_key = self.private_key_options.extract_private_key(
+            self.encoding_options.encoding,
+            &self.profile_options.profile,
+        )?;
+
+        client
+            .create_stake_pool(
+                &network_identifier,
+                &private_key,
```

Comment:
Is copying and passing the private key here safe?

Reply:
This is for tests & stuff only.  It's the same flow as the Aptos CLI

Reply:
Okay as long as this wouldn't be used with mainnet keys.